web/src/app/donate/DonateContainer.js

```diff
@@ -1,18 +1,44 @@
+import * as donateTypes from './duck/types';
 import React, { Component } from 'react';
 import DonateComponent from './DonateComponent';
+import ErrorComponent from './ErrorComponent';
+import InvalidUserComponent from '../common/InvalidUser/InvalidUserComponent';
+import SuccessComponent from './SuccessComponent';
+import LoadingComponent from '../common/loading/LoadingComponent';
 import { bindActionCreators } from 'redux';
 import { connect } from 'react-redux';
 import { donateOperations } from './duck';
+import { publicUserOperations } from '../user/duck';
 
 class DonateContainer extends Component {
 
-  // TODO: make sure this user exists
-  render() {
+  componentWillMount() {
     const twitchUserName = this.props.match.params.twitchUserName;
+    this.props.getPublicUser(twitchUserName);
+  }
+
+  render() {
+    const user = JSON.parse(sessionStorage.getItem('user'));
+    const streamerUserName = this.props.match.params.twitchUserName;
+
+    if (this.props.publicUser.getPublicUserLoading || this.props.donate.curDonationState === donateTypes.DONATION_PROCESSING) {
+      return <LoadingComponent />
+    } else if (this.props.publicUser.publicUser === null) {
+      return <InvalidUserComponent />;
+    } else if (this.props.donate.curDonationState === donateTypes.DONATION_PROCESSED) {
+      return (
+        <SuccessComponent
```

Comment:
same 'should we make this generic' question as the ErrorComponent and the InvalidUserComponent

Reply:
I think Success/Error should be generic, but I think InvalidUserComponent should be its own thing. The reason being, for other resources (challenges, etc) I don't think we'll necessarily want to display a full page error component.